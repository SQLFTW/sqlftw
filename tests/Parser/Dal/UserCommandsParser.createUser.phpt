<?php declare(strict_types = 1);

namespace SqlFtw\Parser;

use SqlFtw\Formatter\Formatter;
use Tester\Assert;

require __DIR__ . '/../../bootstrap.php';

$parser = ParserHelper::getParserFactory()->getParser();
$formatter = new Formatter($parser->getSettings());


// CREATE USER [IF NOT EXISTS] user [auth_option] [, user [auth_option]] DEFAULT ROLE {NONE | ALL | role [, role ] ...}
$query = "CREATE USER 'foo'@'localhost' DEFAULT ROLE admin";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));

$query = "CREATE USER 'foo'@'localhost' DEFAULT ROLE admin, developer";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));

$query = "CREATE USER IF NOT EXISTS 'foo'@'localhost' DEFAULT ROLE admin";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));


// IDENTIFIED BY 'auth_string' [REPLACE 'current_auth_string'] [RETAIN CURRENT PASSWORD]
$query = "CREATE USER 'foo'@'localhost' IDENTIFIED BY 'bar' DEFAULT ROLE admin";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));

$query = "CREATE USER 'foo'@'localhost' IDENTIFIED BY 'bar' REPLACE 'baz' DEFAULT ROLE admin";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));

$query = "CREATE USER 'foo'@'localhost' IDENTIFIED BY 'bar' REPLACE 'baz' RETAIN CURRENT PASSWORD DEFAULT ROLE admin";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));

$query = "CREATE USER 'foo'@'localhost' IDENTIFIED BY 'bar' RETAIN CURRENT PASSWORD DEFAULT ROLE admin";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));


// IDENTIFIED WITH auth_plugin
$query = "CREATE USER 'foo'@'localhost' IDENTIFIED WITH bar DEFAULT ROLE admin";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));


// IDENTIFIED WITH auth_plugin BY 'auth_string' [REPLACE 'current_auth_string'] [RETAIN CURRENT PASSWORD]
$query = "CREATE USER 'foo'@'localhost' IDENTIFIED WITH bar BY 'baz' DEFAULT ROLE admin";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));

$query = "CREATE USER 'foo'@'localhost' IDENTIFIED WITH bar BY 'baz' REPLACE 'bak' DEFAULT ROLE admin";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));

$query = "CREATE USER 'foo'@'localhost' IDENTIFIED WITH bar BY 'baz' REPLACE 'bak' RETAIN CURRENT PASSWORD DEFAULT ROLE admin";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));

$query = "CREATE USER 'foo'@'localhost' IDENTIFIED WITH bar BY 'baz' RETAIN CURRENT PASSWORD DEFAULT ROLE admin";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));


// IDENTIFIED WITH auth_plugin AS 'hash_string'
$query = "CREATE USER 'foo'@'localhost' IDENTIFIED WITH bar AS 'baz' DEFAULT ROLE admin";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));


// more users
$query = "CREATE USER 'foo'@'localhost' IDENTIFIED WITH bar, 'bar'@'localhost' IDENTIFIED WITH bar DEFAULT ROLE admin";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));


// [REQUIRE {NONE | tls_option [[AND] tls_option] ...}]
$query = "CREATE USER 'foo'@'localhost' DEFAULT ROLE admin REQUIRE NONE";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));

$query = "CREATE USER 'foo'@'localhost' DEFAULT ROLE admin REQUIRE SSL";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));

$query = "CREATE USER 'foo'@'localhost' DEFAULT ROLE admin REQUIRE X509";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));

$query = "CREATE USER 'foo'@'localhost' DEFAULT ROLE admin REQUIRE CIPHER 'bar'";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));

$query = "CREATE USER 'foo'@'localhost' DEFAULT ROLE admin REQUIRE ISSUER 'bar'";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));

$query = "CREATE USER 'foo'@'localhost' DEFAULT ROLE admin REQUIRE SUBJECT 'bar'";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));

$query = "CREATE USER 'foo'@'localhost' DEFAULT ROLE admin REQUIRE SSL AND ISSUER 'bar'";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));


// [WITH resource_option [resource_option] ...]
$query = "CREATE USER 'foo'@'localhost' DEFAULT ROLE admin WITH MAX_QUERIES_PER_HOUR 10";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));

$query = "CREATE USER 'foo'@'localhost' DEFAULT ROLE admin WITH MAX_UPDATES_PER_HOUR 10";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));

$query = "CREATE USER 'foo'@'localhost' DEFAULT ROLE admin WITH MAX_CONNECTIONS_PER_HOUR 10";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));

$query = "CREATE USER 'foo'@'localhost' DEFAULT ROLE admin WITH MAX_USER_CONNECTIONS 10";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));

$query = "CREATE USER 'foo'@'localhost' DEFAULT ROLE admin WITH MAX_QUERIES_PER_HOUR 10 MAX_USER_CONNECTIONS 10";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));


// [password_option | lock_option] ...
$query = "CREATE USER 'foo'@'localhost' DEFAULT ROLE admin PASSWORD EXPIRE DEFAULT";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));

$query = "CREATE USER 'foo'@'localhost' DEFAULT ROLE admin PASSWORD EXPIRE NEVER";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));

$query = "CREATE USER 'foo'@'localhost' DEFAULT ROLE admin PASSWORD EXPIRE INTERVAL 365 DAY";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));

$query = "CREATE USER 'foo'@'localhost' DEFAULT ROLE admin PASSWORD HISTORY DEFAULT";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));

$query = "CREATE USER 'foo'@'localhost' DEFAULT ROLE admin PASSWORD HISTORY 2";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));

$query = "CREATE USER 'foo'@'localhost' DEFAULT ROLE admin PASSWORD REUSE INTERVAL DEFAULT";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));

$query = "CREATE USER 'foo'@'localhost' DEFAULT ROLE admin PASSWORD REUSE INTERVAL 365 DAY";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));

$query = "CREATE USER 'foo'@'localhost' DEFAULT ROLE admin PASSWORD REQUIRE CURRENT";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));

$query = "CREATE USER 'foo'@'localhost' DEFAULT ROLE admin PASSWORD REQUIRE CURRENT DEFAULT";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));

$query = "CREATE USER 'foo'@'localhost' DEFAULT ROLE admin PASSWORD REQUIRE CURRENT OPTIONAL";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));


// ACCOUNT LOCK | ACCOUNT UNLOCK
$query = "CREATE USER 'foo'@'localhost' DEFAULT ROLE admin ACCOUNT LOCK";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));

$query = "CREATE USER 'foo'@'localhost' DEFAULT ROLE admin ACCOUNT UNLOCK";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));


// more
$query = "CREATE USER 'foo'@'localhost' DEFAULT ROLE admin PASSWORD EXPIRE DEFAULT ACCOUNT UNLOCK";
Assert::same($query, $parser->parseCommand($query)->serialize($formatter));
