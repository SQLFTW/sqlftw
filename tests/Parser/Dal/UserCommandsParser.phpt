<?php declare(strict_types = 1);

namespace SqlFtw\Parser;

use SqlFtw\Tests\Assert;

require __DIR__ . '/../../bootstrap.php';


// CREATE ROLE [IF NOT EXISTS] role [, role ] ...
Assert::parse("CREATE ROLE role1");
Assert::parse("CREATE ROLE role1, role2");
Assert::parse("CREATE ROLE IF NOT EXISTS role1");


// DROP ROLE [IF EXISTS] role [, role ] ...
Assert::parse("DROP ROLE role1");
Assert::parse("DROP ROLE role1, role2");
Assert::parse("DROP ROLE IF EXISTS role1");


// DROP USER [IF EXISTS] user [, user] ...
Assert::parse("DROP USER usr1@host1");
Assert::parse("DROP USER usr1@host1, usr2@host2");
Assert::parse("DROP USER IF EXISTS usr1@host2");


// RENAME USER old_user TO new_user [, old_user TO new_user] ...
Assert::parse("RENAME USER usr1@host1 TO usr2@host2");
Assert::parse("RENAME USER usr1@host1 TO usr2@host2, usr3@host3 TO usr4@host4");


// SET DEFAULT ROLE {NONE | ALL | role [, role ] ...} TO user [, user ] ...
Assert::parse("SET DEFAULT ROLE NONE TO usr1@host1");
Assert::parse("SET DEFAULT ROLE ALL TO usr1@host1");
Assert::parse("SET DEFAULT ROLE role1 TO usr1@host1");
Assert::parse("SET DEFAULT ROLE role1, role2 TO usr1@host1");
Assert::parse("SET DEFAULT ROLE role1 TO usr1@host1, usr2@host2");


// SET PASSWORD [FOR user] = {PASSWORD('auth_string') | 'auth_string'}
Assert::parse("SET PASSWORD = 'pwd1'");
Assert::parse("SET PASSWORD FOR usr1@host1 = 'pwd1'");
Assert::parse("SET PASSWORD = PASSWORD('pwd1')");


// SET ROLE {DEFAULT | NONE | ALL | ALL EXCEPT role [, role ] ... | role [, role ] ... }
Assert::parse("SET ROLE DEFAULT");
Assert::parse("SET ROLE NONE");
Assert::parse("SET ROLE ALL");
Assert::parse("SET ROLE ALL EXCEPT role1");
Assert::parse("SET ROLE ALL EXCEPT role1, role2");
Assert::parse("SET ROLE role1");
Assert::parse("SET ROLE role1, role2");
