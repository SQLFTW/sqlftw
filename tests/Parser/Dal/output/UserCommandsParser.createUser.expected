CREATE USER 'foo'@'localhost' DEFAULT ROLE admin WITH MAX_QUERIES_PER_HOUR 10 MAX_USER_CONNECTIONS 10